Code/GraphMol/RGroupDecomposition/RGroupDecomp.cpp

```diff
@@ -221,11 +242,19 @@ int RGroupDecomposition::add(const ROMol &inmol) {
             int rlabel = attachments[attach_idx];
             ADD_MATCH(match, rlabel);
             match[rlabel]->add(newMol, attachments);
+#ifdef VERBOSE
+            std::cerr << "Fragment " << i << " " << MolToSmiles(*newMol)
+                      << std::endl;
+#endif
           }
         } else {
           // special case, only one fragment
           if (fragments.size() == 1) {  // need to make a new core
             // remove the sidechains
+            
+            // GJ I think if we ever get here that it's really an error and I believe
```

Comment:
If
1) this code was probed by unit tests previously and it is not anymore after your changes
2) those unit tests still pass after your changes
I'd be in favor of deleting it: the RGD code is already sufficiently complicated :-)

Reply:
This code path was used by tests such as testRingMatching3.  It is no longer used by any of the tests.  I'll leave it in for now and see if @greglandrum has any thought on this, since I think he wrote the code.